### ISSUE TITLE
Differential download fails possible when productName != package name

### ISSUE THREAD

**Issue by danelkhen** [electron-userland/electron-builder#3451]
20.31.2

Done

3.1.6

Windows using NSIS

While testing autoUpdate, I've noticed that differentialDownload fails with ERR_CHECKSUM_MISMATCH. (sha512 checksum mismatch)
After debugging this for a while, I noticed that the installer has created 2 folders in userData, one with the package.json name which contained logs and __installer.exe, and another with the productName (that is different in my case), with the autoUpdate data (__update__, blob_storage, Cache, etc...).
I've played around with both folders, and noticed that if I move the __installer.exe file to the [package.json name] folder, it all works.

Still not sure why and how to explain it, but can you take a look?

To recap, this is how the file structure looked:
```
[user]/AppData/Romaing/[productName]
    logs
    __installer.exe
[user]/AppData/Romaing/[packageName]
    __update__
    blob_storage
    Cache
```

**Comment by danelkhen** [comment]
update: wrong folder name is caused by app.getName() returns package.json.name, when productName is not set in package.json and only set in the electron-builder.json file.

**Comment by danelkhen** [comment]
Verified: issue is caused by having `electron-builder.json[productName]` that differs from `package.json[name]` *and* not having a `package.json[productName]`

To fix the issue:
Set` productName` in `package.json` (root element, sibling of `name`)
Remove the `productName` from `electron-builder.json` (optional, if remains it should match `package.json.productName`)

I would suggest to warn about this during electron-builder run

**Comment by zwdgit** [comment]
In which version solved this problem？

**Comment by alexandermckay** [comment]
I am also getting this error. Has anyone had any success dealing with it?
